client/packages/system/src/Stock/Components/Repack/RepackModal.tsx

```diff
@@ -211,10 +208,21 @@ export const RepackModal: FC<RepackModalControlProps> = ({
           <Box paddingLeft={3} paddingTop={3} flex={1}>
             {showRepackDetail && (
               <RepackEditForm
-                invoiceId={invoiceId}
                 onChange={onChange}
-                stockLine={stockLine}
-                draft={draft}
+                data={{
+                  stockLineId: draft?.stockLineId ?? stockLine?.id,
+                  packSize: draft?.packSize ?? stockLine.packSize,
+                  newPackSize: draft?.newPackSize,
+                  numberOfPacks:
+                    draft?.numberOfPacks ?? repackData?.to?.numberOfPacks,
+                  newLocationId: draft?.newLocationId,
+                  availableNumberOfPacks:
+                    draft?.availableNumberOfPacks ??
+                    stockLine?.availableNumberOfPacks,
+                  locationName:
+                    draft?.locationName ?? repackData?.to?.location?.name,
```

Comment:
Oh hm, I didn't realise how the different inputs mapped 😅 

Also looks like data for viewing existing repacks isn't mapping quite right...
![Screenshot 2024-06-07 at 4 40 51 PM](https://github.com/msupply-foundation/open-msupply/assets/55115239/756ae792-4785-4940-9392-d939283b5096)

I had a play here, wonder what you think? https://github.com/msupply-foundation/open-msupply/compare/3656-(3)-finish-stock-API-refactor...3656-(3)-finish-stock-API-refactor(form-data)

Reply:
That's great, thanks.